lib/Makefile

```diff
@@ -223,42 +222,78 @@ lib-all: all
 
 all: lib
 
-$(ZSTD_STATLIB_DIR)/libzstd.a: ARFLAGS = rcs
-$(ZSTD_STATLIB_DIR)/libzstd.a: | $(ZSTD_STATLIB_DIR)
-$(ZSTD_STATLIB_DIR)/libzstd.a: $(ZSTD_STATLIB_OBJ)
+ifeq ($(BUILD_DIR),0)
+# determine a BUILD_DIR
+
+.PHONY: libzstd.a  # not the actual recipe
+libzstd.a:
+	$(MAKE) --no-print-directory $@ \
+        BUILD_DIR=obj/$(HASH_DIR) \
+        CPPFLAGS="$(CPPFLAGS)"
+
+else
+# $(BUILD_DIR) is defined
+
+ZSTD_STATLIB_DIR := $(BUILD_DIR)/static
+ZSTD_STATLIB := $(ZSTD_STATLIB_DIR)/libzstd.a
+ZSTD_STATLIB_OBJ := $(addprefix $(ZSTD_STATLIB_DIR)/,$(ZSTD_LOCAL_OBJ))
+$(ZSTD_STATLIB): ARFLAGS = rcs
+$(ZSTD_STATLIB): | $(ZSTD_STATLIB_DIR)
```

Comment:
What does this line do?

Reply:
It creates the target directory if it doesn't exist